utils/jwe.ts

```diff
@@ -51,3 +66,63 @@ export const getGenerateJWE: GetGenerateJWE = (issuer, jweKey) => (
       TE.fromPredicate(NonEmptyString.is, () => E.toError("Token is empty."))
     )
   );
+
+/**
+ * JWE Validation
+ */
+type GetValidateJWE = (
+  issuer: NonEmptyString,
+  jwePrivateKey: NonEmptyString
+) => (token: NonEmptyString) => TE.TaskEither<Error, BaseJwePayload>;
+
+const getPayloadFromDecryptResult = (
+  cryptedPayload: jose.CompactDecryptResult
+): TE.TaskEither<Error, BaseJwePayload> =>
+  pipe(
+    E.tryCatch(
+      () => new TextDecoder().decode(cryptedPayload.plaintext),
+      E.toError
+    ),
+    E.chain(J.parse),
+    E.chainW(BaseJwePayload.decode),
+    TE.fromEither,
+    TE.mapLeft(_ => E.toError(`Could not decode JWE`))
+  );
+
+const validateIssAndExp = (
+  payload: BaseJwePayload,
+  issuer: NonEmptyString
+): TE.TaskEither<Error, BaseJwePayload> =>
+  pipe(
+    payload,
+    TE.fromPredicate(
+      x => issuer === x.iss,
+      () => new Error("Issuer does not match")
+    ),
+    TE.chain(
+      TE.fromPredicate(
+        x => x.exp > Math.floor(Date.now() / 1000),
+        () => new Error("Token expired")
+      )
+    )
+  );
+
+export const validateJweWithKey = (
+  token: NonEmptyString,
+  jwePrivateKey: NonEmptyString,
+  issuer: NonEmptyString
+): TE.TaskEither<Error, BaseJwePayload> =>
+  pipe(
+    getKeyLikeFromString(jwePrivateKey),
+    TE.chain(pkcs8 =>
+      TE.tryCatch(() => jose.compactDecrypt(token, pkcs8), E.toError)
+    ),
+    TE.mapLeft(e => new Error(`Error decrypting Magic Link JWE: ${e}`)),
+    TE.chain(crypted => getPayloadFromDecryptResult(crypted)),
+    TE.chain(payload => validateIssAndExp(payload, issuer))
+  );
```

Comment:
If we use `jwtDecrypt` method the most of the validation is done by the library itself.

```suggestion
export const validateJweWithKey = (
  token: NonEmptyString,
  jwePrivateKey: NonEmptyString,
  issuer: NonEmptyString
): TE.TaskEither<Error, BaseJwePayload> =>
  pipe(
    getKeyLikeFromString(jwePrivateKey),
    TE.chain(pkcs8 =>
      TE.tryCatch(() => jose.jwtDecrypt(token, pkcs8, { issuer }), E.toError)
    ),
    TE.mapLeft(e => new Error(`Error decrypting Magic Link JWE: ${e}`)),
    TE.chain(
      flow(
        BaseJwePayload.decode,
        E.mapLeft(
          err =>
            new Error(`Invalid JWE payload: ${readableReportSimplified(err)}`)
        ),
        TE.fromEither
      )
    )
  );
```